projects/media-viewer/src/lib/toolbar/left-pane/left-pane.component.ts

```diff
@@ -18,10 +19,41 @@ export class ToolbarLeftPaneComponent {
   @Input() showNavigationBtns: boolean;
   @Input() sidebarOpen: BehaviorSubject<boolean>;
   @Input() searchBarHidden: BehaviorSubject<boolean>;
-  @Input() drawMode: BehaviorSubject<boolean>;
-  @Input() highlightMode: BehaviorSubject<boolean>;
+  // Local presentation state that can be pre-initialised from the template
+  @Input() highlightMode = false;
+  @Input() drawMode = false;
+
+  // local array of any subscriptions so that we can tidy them up later
+  private subscriptions: Subscription[] = [];
+  constructor(private readonly toolbarEventsService: ToolbarEventsService) {}
+
+  ngOnInit(): void {
+    // Listen for any changes invoked on the toolbar events Service and initialise any default behaviour state
+    this.subscriptions.push(this.toolbarEventsService.highlightMode.subscribe((toggleValue) => {
```

Comment:
I would re-iterate my objection from our call earlier.... it seems we're adding a lot of boiler plate (that is will duplicated in other classes too) in  order to avoid `| async`.

Reply:
I hear you, but on the contrary, i disagree in that. I believe it makes it far clearer that we are listening for key events, and whilst its more explicit, it does make it easier to debug and walk through the logic.

In this instance, its changing an internal property that is only being used by the view template. So this is a good candidate to use the async pipe…

Reply:
i’ve conceded in this example… its now an async pipe. @linusnorton can you recheck?